tests/validation/cattlevalidationtest/core/test_github.py

```diff
@@ -44,32 +45,122 @@ def test_github_token_has_access():
     redirect_url = r.headers['location']
     query = urlparse.urlparse(redirect_url)[4]
     query = urlparse.parse_qs(query)
+    driver.quit()
+    return query['code']
 
+
+@pytest.fixture(scope='module')
+def github_request_token(github_request_code, cattle_url):
+    code = github_request_code
+
+    URL = cattle_url
     BASE_URL = URL[:-1 * len('schemas')]
 
-    c = requests.post(BASE_URL + 'token', {'code': query['code']})
+    c = requests.post(BASE_URL + 'token', {'code': code})
+
+    return c.json()['jwt']
 
-    jwt = c.json()['jwt']
 
+@if_github
+def test_github_auth_config_unauth_user(github_request_token):
+    #   switch on auth
     requests.post(BASE_URL + 'githubconfig',
                   data=json.dumps({'enabled': 'true'}))
 
+#   do not set any auth headers
     no_auth = requests.get(URL)
 
+#   test that auth is switched on
     assert no_auth.status_code == 401
 
+    jwt = github_request_token
+#   switch it back
+    requests.post(BASE_URL + 'githubconfig',
+                  headers={'Authorization': 'Bearer ' + jwt},
+                  data=json.dumps({'enabled': 'false'}))
+
+
+@if_github
+def test_github_auth_config_invalid_user(github_request_token):
+    #   switch on auth
+    requests.post(BASE_URL + 'githubconfig',
+                  data=json.dumps({'enabled': 'true'}))
+
+#   set invalid auth headers
     bad_auth = requests.get(URL,
                             headers={'Authorization':
                                      'Bearer some_random_string'})
 
+#   test that user does not have access
     assert bad_auth.status_code == 401
 
+    jwt = github_request_token
+#   switch it back
+    requests.post(BASE_URL + 'githubconfig',
```

Comment:
Since this same basic line appears many times I would probably make a helper function like `def update_github_config(data)`.